fuse_models/include/fuse_models/common/sensor_proc.h

```diff
@@ -368,12 +368,19 @@ inline bool processAbsolutePoseWithCovariance(
  * pose_relative = pose_absolute1^-1 * pose_absolute2
  *
  * Additionally, the covariance of each pose message is rotated into the robot's base frame at the time of
- * pose_absolute1. They are then added in the constraint. This assumes independence between the pose measurements.
+ * pose_absolute1. They are then added in the constraint if the pose measurements are independent.
+ * Otherwise, if the pose measurements are dependent, the covariance of pose_absolute1 is substracted from the
```

Comment:
I've found covariance subtraction to be...challenging. Haven't read the code yet, though.

Reply:
I'm essentially doing the inverse of the common covariance propagation of `p2 = p1 * dp`, where `p1` and `p2` are two consecutive absolute poses and `dp` is the incremental/differential/relative pose.

That would have a covariance `\Sigma_2 = J_p1 * \Sigma_1 * J_p1^T + J_dp * \Sigma_dp * J_dp`. Here I'm simply moving `\Sigma_dp` to the left hand side. This gives better results than simply changing the `+` into `-` for the equation that was already here, for the covariance, that has the jacobians for `dp = p1^-1 * p2`, not `p2 = p1 * dp`.

Anyway, for the `fuse_models::Odometry2D` sensor model, the approach in https://github.com/locusrobotics/fuse/pull/138 is preferred. This would be an option for the `fuse_models::Imu2D` and `fuse_models::Pose2D` sensor models, because their measurements don't have a **twist** (covariance).